hadoop-client-modules/hadoop-client-check-invariants/src/test/resources/ensure-jars-have-correct-contents.sh

```diff
@@ -67,6 +67,8 @@ allowed_expr+="|^krb5_udp-template.conf$"
 # Jetty uses this style sheet for directory listings. TODO ensure our
 # internal use of jetty disallows directory listings and remove this.
 allowed_expr+="|^jetty-dir.css$"
+# Snappy java is native library. We cannot relocate it to under org/apache/hadoop.
```

Comment:
Does this mean it could potentially conflict with downstream's dependency? such as the one from `parquet-mr`?

Reply:
It is possible, yes. I don't have better idea. Actually I remember when we checked dependencies at the snappy codec PR, hadoop-common already has transitive dependency on snappy-java before it. So we don't make it as `provided` like lz4-java.

Reply:
One possibility, is to change to snappy-java to use `RegisterNatives` to registers native methods. Then we might be able to better bind native methods to Java under relocation situation. We can work on this direction later.

Reply:
What is `RegisterNatives`? is it something exists in snappy-java?

Reply:
By default, JNI uses fully qualified class name to resolve native methods. Alternatively, JNI provides an API `RegisterNatives` used to register native methods that don't require to match with class name.